### PR TITLE
Minor fixes for MaskFormer

### PR DESCRIPTION
**Pull request by FrancescoSaverioZuppichini**
# What does this PR do?

Fix a bug found by @Narsil  in the `feature_extractor.post_process_panoptic_segmentation`

### CODE REVIEW

File: tests/maskformer/test_modeling_maskformer.py [hunk 1]
```diff
@@ -403,3 +403,23 @@ def test_with_annotations_and_loss(self):
...
+            ],
+            return_tensors="pt",
+        )
+
+        with torch.no_grad():
+            outputs = model(**inputs)
+
+        panoptic_segmentation = feature_extractor.post_process_panoptic_segmentation(outputs)
+
+        self.assertTrue(len(panoptic_segmentation) == 2)
```
Comment (Narsil) [inline tests/maskformer/test_modeling_maskformer.py:425]: Not sure what's the best approach, but testing against real values seems better if you can easily craft one.

Like maybe we can craft (1, 2, 2) tensors that simulate a model output on 2x2 images and make sure everything including the masks.

Something else if crafting the values is too tedious/brittle, in the pipeline I use  nested_simplify to enable doing that:

```python
self.assertEqual(panoptic_segmentation, [{"segmentation": [[1, 2,3 ]], "segments": [{"id":1, "category_id": ...}, {...}, ....]},
 {"segmentation": [[1, 2,3 ]], "segments": [{"id":1, "category_id": ...}, {...}, ....]}])
```

It currently better than no test. And if the testing is hard to make more complete, we don't necessarily have to.
Tests are relevant as long as they are readable IMO, so a complex tests looses value.
This one is simple and that's good.

Not sure if this applies here and so forth, but I think the test could check a little more than the outer length format IMO, is what I was trying to convey

**Review by Narsil — COMMENTED** [review]
Thanks for this, not approving since this is not my area of competence, but it looks good.

File: tests/maskformer/test_modeling_maskformer.py (hunk 1 above)
Comment (FrancescoSaverioZuppichini) [reply tests/maskformer/test_modeling_maskformer.py:426]: You are definitely correct, I'll have to manually send an image thought maskformer and record some of the outputs

**Review by sgugger — APPROVED** [review]
Thanks for fixing!
